R/Renderer.R

```diff
@@ -33,20 +33,42 @@ Renderer <- R6::R6Class( # nolint: object_name_linter.
       if (missing(yaml_header)) {
         yaml_header <- md_header(yaml::as.yaml(list(title = "Report")))
       }
-      parsed_yaml <- yaml_header
+      report_type <- reverse_yaml_field(yaml_header, "output")
       parsed_global_knitr <- sprintf(
-        "\n```{r setup, include=FALSE}\nknitr::opts_chunk$set(%s)\n```\n",
-        capture.output(dput(global_knitr))
+        "\n```{r setup, include=FALSE}\nknitr::opts_chunk$set(%s)\n%s\n```\n",
+        capture.output(dput(global_knitr)),
+        if (identical(report_type, "powerpoint_presentation")) {
+          paste(
+            "# additional function to parse code block into a table for PPT",
+            "library('flextable')",
+            "code_block <- function(code_block) {",
+            "  df <- data.frame(code_block)",
+            "  ft <- flextable::flextable(df)",
+            "  ft <- flextable::delete_part(ft, part = 'header')",
+            "  ft <- flextable::autofit(ft, add_h = 0)",
+            "  ft <- flextable::fontsize(ft, size = 7, part = 'body')",
+            "  ft <- flextable::bg(x = ft,bg = 'lightgrey')",
+            "  ft <- flextable::border_outer(ft)",
+            "  if(flextable_dim(ft)$widths > 8) {",
+            "    ft <- width(ft, width = 8)",
+            "  }",
+            "  ft",
+            "}",
+            collapse = "\n"
```

Comment:
Collapse only applies for arguments that are vectors `paste(1:10, collapse = "\n")` otherwise, `sep` is needed `paste(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, sep="\n")`

```suggestion
            sep = "\n"
```